edgelet/doc/devguide.md

```diff
@@ -1,163 +1,264 @@
+    Install it with:
 
-#### Code generation
+    ```sh
+    cd edgelet/
 
-We use a modified version of `swagger-codegen` to generate code from our swagger definitions. To build the tool:
+    rustup component add clippy
+    ```
 
-```
-git clone -b support-rust-uds  https://github.com/avranju/swagger-codegen.git
-cd swagger-codegen
-mvn clean package
-```
+    Run it with:
 
-To run the tool, for example to update our workload API:
+    ```sh
+    cargo clippy --all
+    cargo clippy --all --tests
+    cargo clippy --all --examples
+    ```
 
-```
-java -jar swagger-codegen-cli.jar generate -i api/workload.yaml -l rust -o {root}/edgelet/workload
-```
+- Swagger
 
-> Note that `cargo fmt` and `cargo clippy` will complain about the code produced by this tool. We like to keep **all** code in our repo clean, so you'll want to run clippy and fmt over the generated code, make the recommended changes, and carefully inspect the diffs of modified files before checking in.
+    Some of our source code is generated from swagger definitions stored as YAML.
 
-## IDE
-VS Code has good support for rust. Consider installing the following extensions:
+    You can edit the definitions in VS code, but https://editor.swagger.io is also an invaluable tool for validation, converting YAML -> JSON for code-gen, etc.
 
-* [Rust](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust) - Syntax highlighting and intellisense support
-* [Better TOML](https://marketplace.visualstudio.com/items?itemName=bungcip.better-toml) - Syntax highlighting for Cargo.toml
-* [C/C++](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) - Native debugger support
-* [Vim](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim) - For a more sophisticated editor experience :)
+    We use a modified version of `swagger-codegen` to generate code from the swagger definitions. To build the tool:
+
+    ```sh
+    git clone -b support-rust-uds https://github.com/avranju/swagger-codegen.git
+    cd swagger-codegen
+    mvn clean package
+    ```
 
-There is a `launch.json` configuration in this repo to setup debugging on Windows. This should work out of the box.
+    To run the tool, for example to update our workload API:
+
+    ```sh
+    java -jar swagger-codegen-cli.jar generate -i api/workload.yaml -l rust -o {root}/edgelet/workload
+    ```
+
+    Note that we've manually fixed up the generated code so that it satisfies rustfmt and clippy. As such, if you ever need to run `swagger-codegen-cli` against new definitions, or need to regenerate existing ones, you will want to perform the same fixups manually. Make sure to run clippy and rustfmt against the new code yourself, and inspect the diffs of modified files before checking in.
+
+- IDE
+
+    VS Code has good support for Rust. Consider installing the following extensions:
```

Comment:
Should we mentioned IntelliJ IDEA with Rust plugin?

Reply:
I don't know anything about it. What do you want it to say?

Reply:
> Alternatively, [Intellij IDEA Community Edition](https://www.jetbrains.com/idea) with [Rust plugin](https://intellij-rust.github.io) provides full IDE experience for programming in Rust.

Something like this maybe?

Reply:
Done.